.codeclimate.yml

```diff
@@ -4,6 +4,28 @@
 
 version: "2"
 
+checks:
+  argument-count:
+    enabled: false
+  complex-logic:
+    enabled: false
+  file-lines:
+    enabled: false
+  method-complexity:
+    enabled: false
+  method-count:
+    enabled: false
+  method-lines:
+    enabled: false
+  nested-control-flow:
+    enabled: false
+  return-statements:
+    enabled: false
+  similar-code:
+    enabled: false
+  identical-code:
+    enabled: false
```

Comment:
Are we sure the best thing to do is disable all these checks? I think it indicates a code smell if there are this many issues that need to be ignored. We usually have lengthy discussions before relaxing even any single rule, to make sure we keep the code maintainable.

Reply:
@rafis3   Could I get more information on why we think this is right approach?  Code climate does provide additional code checks that seem beneficial from a code quality POV.

Reply:
Yes of course. I was in the middle of some experimenting that didn't totally work.
We currently run Pylint as part of the pipeline. Also during development, we run it to make sure it's 10/10 at all time.
Sometimes you need to exclude a certain rule in a certain place, because you have a justification for it. You do it using a comment that looks like this:

```python
# pylint: disable=too-few-public-methods
```

While Pylint respects these comments, CodeClimate ignores them. I couldn't yet figure out why, but that causes a lot of issues, even though they are supposed to be ignored.

So ideally, I would love to keep all the CodeClimate rules, just tell it to ignore the Python code, because Pylint already covers us there.